Services/PriceProvider.cs

```diff
@@ -30,13 +41,35 @@ public async Task AddPrices(IList<ShoppingCartItem> items)
             {
                 if (skuProducts.TryGetValue(item.ProductSku, out var product))
                 {
-                    var pricePart = product.ContentItem.As<PricePart>();
-                    if (pricePart != null)
+                    ContentItem contentItem = product.ContentItem;
+
+                    foreach (var pricePart in PartsOfType<PricePart>(contentItem))
                     {
-                        item.Prices.Add(new PrioritizedPrice(0, pricePart.Price));
+                        if (pricePart.Price.Currency == _moneyService.CurrentDisplayCurrency)
+                        {
+                            item.Prices.Add(new PrioritizedPrice(0, pricePart.Price));
+                        }
                     }
                 }
             }
         }
+
+        public IEnumerable<TPart> PartsOfType<TPart>(ContentItem contentItem) where TPart : ContentPart
+        {
+            var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(contentItem.ContentType);
+
+            foreach (var contentTypePartDefinition in contentTypeDefinition.Parts)
```

Comment:
This can be one simple `Select`, `Where` and `Cast<TPart>`

Reply:
This change was to make it work with named parts. I didn't got it to work in any other way.

I had issues with the ContentItem extensions methods named parts.
For example: contentItem.As<PricePart>(); don't return anything for a named part. It only gets a match if the name is the same as the part name.

I'll be doing some more investigation in this area and check with the Orchard Core team.

The original code that was before this didn't work with named parts. I think it's a good reason for adding a test product with this configuration so changes are always tested with it.

Reply:
You can retrieve named parts using something like this:
```
contentItem.Get<ContentPart>("MyContentPart")
```
Replace "ContentPart" with the type of part you are retrieving (or keep "ContentPart" if it is a generic part). Obviously, replace "MyContentPart" with the named part "Name".

Reply:
Thanks. I'll try to implement a variant of this that will make it possible to call like the following:
var parts = contentItem.OfType<PricePart>();

I'll do it as an extension method so that the dependency on ContentDefinitionManager can be removed. The idea is to get all parts of a specific type without having to know the name.

Reply:
Simplified the helper method PartsOfType a litte bit. Have also written an extension method for ContentItem which will make it possible to remove the dependency on ContentDefinitionManager and the helper method PartsOfType completely.

```csharp
        /// <summary>
        /// Gets all content elements of a specific type.
        /// </summary>
        /// <typeparam name="TElement">The expected type of the content elements.</typeparam>
        /// <returns>The content element instances or empty sequence if no entries exist.</returns>
        public static IEnumerable<TElement> OfType<TElement>(this ContentElement contentElement) where TElement : ContentElement
        {
            foreach (var part in contentElement.Elements)
            {
                if (part.Value is TElement)
                {
                    var result = part.Value as TElement;

                    yield return result;
                }
            }
        }
```